packages/app-store/_utils/nodeScheduler.ts

```diff
@@ -0,0 +1,298 @@
+import type { Prisma } from "@prisma/client";
+import schedule from "node-schedule";
+import { v4 } from "uuid";
+
+import { getHumanReadableLocationValue } from "@calcom/core/location";
+import { getCalEventResponses } from "@calcom/features/bookings/lib/getCalEventResponses";
+import { getTranslation } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+import type { ApiKey } from "@calcom/prisma/client";
+import { BookingStatus, WebhookTriggerEvents } from "@calcom/prisma/enums";
+
+export async function addSubscription({
+  appApiKey,
+  triggerEvent,
+  subscriberUrl,
+  appId,
+}: {
+  appApiKey: ApiKey;
+  triggerEvent: WebhookTriggerEvents;
+  subscriberUrl: string;
+  appId: string;
+}) {
+  try {
+    const createSubscription = await prisma.webhook.create({
+      data: {
+        id: v4(),
+        userId: appApiKey.userId,
+        teamId: appApiKey.teamId,
+        eventTriggers: [triggerEvent],
+        subscriberUrl,
+        active: true,
+        appId: appId,
+      },
+    });
+
+    if (triggerEvent === WebhookTriggerEvents.MEETING_ENDED) {
+      //schedule job for already existing bookings
+      const where: Prisma.BookingWhereInput = {};
+      if (appApiKey.teamId) where.eventType = { teamId: appApiKey.teamId };
+      else where.userId = appApiKey.userId;
+      const bookings = await prisma.booking.findMany({
+        where: {
+          ...where,
+          startTime: {
+            gte: new Date(),
+          },
+          status: BookingStatus.ACCEPTED,
+        },
+      });
+
+      for (const booking of bookings) {
+        scheduleTrigger(booking, createSubscription.subscriberUrl, {
+          id: createSubscription.id,
+          appId: createSubscription.appId,
+        });
+      }
+    }
+
+    return createSubscription;
+  } catch (error) {
+    return console.error(
+      `Error creating subscription for user ${appApiKey.userId} and appId ${appApiKey.appId}.`
+    );
+  }
+}
+
+export async function deleteSubscription({
+  appApiKey,
+  webhookId,
+  appName,
```

Comment:
nit: We should use same name for the same entity. `addSubscription` calls it `appId`. I like that name better.